assets/chat/js/autocomplete.js

```diff
@@ -130,7 +130,7 @@ class ChatAutoComplete {
       } else if (char.length > 0) {
         this.promoteIfSelected();
         const str = this.input.val().toString();
-        const offset = this.input[0].selectionStart + 1;
+        const offset = this.input[0].selectionStart;
         const pre = str.substring(0, offset);
         const post = str.substring(offset);
         const criteria = buildSearchCriteria(pre + char + post, offset);
```

Comment:
Removing the `+ 1` fixes the issue, but introduces a new bug where the completion suggestions don't take the final letter of the word into account. It seems `buildSearchCriteria()` expects an offset representing the new caret position with the new character.

```suggestion
        const criteria = buildSearchCriteria(pre + char + post, offset + 1);
```